core/api/storage_miner/storage_api.cpp

```diff
@@ -79,11 +79,54 @@ namespace fc::api {
       return result;
     };
 
-    api->SectorsStatus = [=](SectorNumber id,
-                             bool) -> outcome::result<ApiSectorInfo> {
-      // TODO(ortyomka): [FIL-421] implement it
+    api->SectorsStatus =
+        [=](SectorNumber id,
+            bool show_onchain_info) -> outcome::result<ApiSectorInfo> {
       OUTCOME_TRY(sector_info, miner->getSealing()->getSectorInfo(id));
-      return ApiSectorInfo{.state = sector_info->state};
+      std::vector<DealId> deals;
+      for (auto &piece : sector_info->pieces) {
+        if (piece.deal_info) {
+          deals.push_back(piece.deal_info->deal_id);
+        }
+      }
+      ApiSectorInfo api_sector_info{
+          sector_info->state,
+          id,
+          sector_info->sector_type,
+          sector_info->comm_d,
+          sector_info->comm_r,
+          sector_info->proof,
+          deals,
+          sector_info->ticket,
+          sector_info->seed,
+          sector_info->precommit_message,
+          sector_info->message,
+          sector_info->invalid_proofs,
+          miner->getSealing()->isMarkedForUpgrade(id)};
```

Comment:
Let's move it to Miner as separate function `sectorStatus(SectorNumber)`

Reply:
circular dependency, we cannot do it